### ISSUE TITLE
Can this tool create images containing molecular bbox or mask coordinates?

### ISSUE THREAD

**Issue by sralvins** [OBrink/RanDepict#55]
I need molecule images with molecular bbox/mask to train a segmetation model.

Can RanDepict create images containing molecular bbox or mask coordinates?

**Comment by Kohulan** [comment]
Hi @sralvins ,

Currently, we don't generate this. This tool is specifically designed to generate chemical structure images in various styles.

**Comment by OBrink** [comment]
Hey @sralvins,

I have written code to produce chemical structure images with binary masks based on RanDepict.
We have used it to create synthetic data that the latest DECIMER Segmentation model has been pretrained with.
Have a look here: https://github.com/OBrink/chemical_page_segmentation_dataset/blob/4c2ad7bdfa1dc0aafb6b85e7699506df3e196662/training_data_generation/ChemSegmentationDatasetCreator.py#L321

Let me know if you need assistance! Have a great day!
Otto

**Comment by sralvins** [comment]
Thank you.

I’ve been trying to reproduce the model proposed in the paper [ChemSAM](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00823-2) (the model is not open-source). The workflow described in the paper seems to combine the SAM model with DECIMER-Image-Segmentation's post-processing operations.

I wonder if the SAM model performs well in chemical molecule segmentation.
